scipy/optimize/_optimize.py

```diff
@@ -3218,9 +3217,11 @@ def _minimize_powell(func, x0, args=(), callback=None, bounds=None,
 
             # Construct the extrapolated point
             direc1 = x - x1
-            x2 = 2*x - x1
             x1 = x.copy()
-            fx2 = squeeze(func(x2))
+            # make sure that we don't go outside the bounds when extrapolating
+            _, lmax = _line_for_search(x, direc1, lower_bound, upper_bound)
```

Comment:
Apologies for not catching this earlier. In the `_line_for_search` function, I believe it expects `lower_bound` to be of the form `[lower_bound_coord1, lower_bound_coord2, ...]`, and same for `upper_bound`. If there is no bound for e.g. coordinate 2, then `lower_bound_coord2` should be `-np.inf` I think. So in this function call we cannot use the `lower_bound` and `upper_bound` as defined earlier in the function, since they are defined to be `None` if there are no bounds.

Reply:
Yeah, I've just been seeing this.

Reply:
Would it be cleaner to do:
```
# Construct the extrapolated point
direc1 = x - x1
x1 = x.copy()
if lower_bound is None and upper_bound is None:
    lmax = 1
else:
    _, lmax = _line_for_search(x, direc1, lower_bound, upper_bound)

x2 = x + min(lmax, 1) * direc1
fx2 = squeeze(func(x2))
```

Reply:
@andyfaff I have tried running with the source code (ver 1.7.3) with the lines above, but it still generates the same result and it performs function evaluation outside of the lower bound.